### ISSUE TITLE
Build 3617+ crashes with

### ISSUE THREAD

**Issue by jonpalmisc** [Vector35/binaryninja-api#3373]
Standard Mach-O file. Crash seems related to the new string-related stuff:

```
* thread #27, stop reason = EXC_BAD_ACCESS (code=1, address=0x8)
  * frame #0: 0x000000010c8d4450 libbinaryninjacore.1.dylib`BinaryNinjaCore::TypeData::GetOffset(this=0x0000000000000000) const at type.cpp:499:6 [opt]
    frame #1: 0x000000010c975fec libbinaryninjacore.1.dylib`BinaryNinjaCore::Unicode::CheckForUTF32Annotation(view=0x0000000104f98600, addr=4296120896, value=0x0000000175ea5c48, allowShortStrings=true, maxLength=32, unicodeBlocks=size=3) at unicode.cpp:836:31 [opt]
    frame #2: 0x000000010c0b7784 libbinaryninjacore.1.dylib`BinaryNinjaCore::Analysis::CheckForStringAnnotation(this=0x0000000144431a00, addr=4296120896, value=0x0000000175ea5c48, allowShortStrings=true) at analysis.cpp:1159:24 [opt]
    frame #3: 0x000000010c6d3550 libbinaryninjacore.1.dylib`BinaryNinjaCore::MediumLevelILTranslator::TranslateExpr(this=0x0000000175ea6800, src=0x00000002ac485a00, dest=0x000000017b270200, ...)
```

[Paletro.zip](https://github.com/Vector35/binaryninja-api/files/9305247/Paletro.zip)

**Comment by comex** [comment]
I'm also seeing a reliable crash when loading a certain dylib on 3.1.3617; let me know if more details are needed.

**Comment by cxnder** [comment]
Resolved in builds >= 3.1.3622
